### ISSUE TITLE
remove readyOnLoad + readyOnInitialize?

### ISSUE THREAD

**Issue by lancejpollard** [segment-boneyard/analytics.js-integrations#251]
Basically this is to simplify testing. The ideal would be something like there are 2 states:
- before loading: testing that options and configuration gets setup
- after loading: testing that events and such get called

But currently there is a special case built in for integrations that have their own queuing system built-in. For example, the google analytics snippet does this:

``` js
window.ga = window.ga || [];
```

so you can do `ga.push` right away, but later on when the library loads it replaces `ga.push` with their own function for making the API call and whatnot. So `readyOnInitialize` is to handle that specific case, so we can let the library start sending data during it's load process (if it processes it right away).

The issue is, this means that there are 2 sort-of "ready" states, there's:
1. ready to push data into a queue
2. ready to push data and have it save

So the way that's implemented now, you basically have to wait until `load`, which is for the second case. But `load` doesn't fire if it's `readyOnInitialize`... so it starts to get super confusing what the actual state of the integration is.

This makes testing hard, because the integration may be `readyOnInitialize`, which means the library supports it's own queuing mechanism, and so it's "ready", but all of the global variables it sets up aren't ready, so `integration.loaded()` is false. In order to accurately test the global variables, which tells you it has successfully loaded, you have to wait until `load`.

It also happens sometimes where, inside the `integration.track` method, we use some info from some of the variables that get loaded and configured by the external library, so even though it's technically `readyOnInitialize` because it has an eager queuing mechanism, it's not possible to `.track` until it's loaded, so then we have to change it to `readyOnLoad`.

So basically, we're thinking of just waiting until the library is fully loaded until it's considered `"ready"`, and remove the `readyOnX` methods, since we handle queuing the calls internally already. This should simplify implementation and testing a bunch.

As an aside, Google Analytics has this eager queuing mechanism, but it's `readyOnLoad` because we use some of its extra info in the `.track` or `.page` calls after the JS library is fully loaded, but technically it's `readyOnInitialize`. So too keep things simple, it'll be nice to just move it all to `ready`, which means the library is fully loaded.

/cc @ianstormtaylor @yields @calvinfo @reinpk @DirtyAnalytics
